### DISCUSSION TITLE
ClayFF data file

### DISCUSSION THREAD

**Discussion by martinhp27** [ISISNeutronMuon/MDMC#733] [Q&A]
Dear all
I'm currently trying to implement ClayFF into MDMC and I think the best way to do so, is to create a data file similar to OPSLAA. 
I found the following data file in lammps github (https://github.com/lammps/lammps/tree/master/tools/msi2lmp/frc_files)

If you have some good ideas how to understand this frc file or change it to the dat file format, please let me know :) 
- Martin

!CLAYFF forcefield

#atom_types	cvff

!Ver  Ref  Type    Mass      Element  Connections   Description and charge
!---- ---  ----  ----------  -------  -----------------------------------------
 1.0   1    st     28.08550    Si          4        tetrahedral silicon  2.1
 1.0   1    ao     26.98154    Al          6        octahedral aluminum  1.575
 1.0   1    at     26.98154    Al          4        tetrahedral aluminum 1.575
 1.0   1   mgo     24.30500    Mg          6        octahedral magnesium 1.36
 1.0   1   mgh     24.30500    Mg          6        hydroxide magnesium  1.05
 1.0   1   cao     40.08000    Ca          6        octahedral calcium   1.36
 1.0   1   cah     40.08000    Ca          6        hydroxide calcium    1.05
 1.0   1   feo     55.84700    Fe          6        octahedral iron      1.575
 1.0   1   lio     6.941000    Li          6        octahedral lithium   0.525
 1.0   1    ob    15.99940      O          2        bridging oxygen     -1.05
 1.0   1  obss     15.99940     O          3        oxygen double sub.  -1.2996
 1.0   1  obts     15.99940     O          2        oxygen tet. sub.    -1.1688
 1.0   1  obos     15.99940     O          2        oxygen oct. sub.    -1.1808
 1.0   1   ohs     15.99940     O          2        hydroxyl O sub.     -1.0808
 1.0   1    oh     15.99940     O          2        hydroxyl O          -0.95
 1.0   1   oh-     15.99940     O          1        hydroxide O
 1.0   1    o*     15.99940     O          2        spc water O         -0.82
 1.0   1    ho     1.007970     H          1        hydroxyl H           0.425
 1.0   1    h*     1.007970     H          1        spc water H          0.41
 1.0   1    Na     22.99000    Na          0        sodium ion           1.0
 1.0   1     K     39.10        K          0        potassium ion        1.0
 1.0   1    Cs     132.9100    Cs          0        cesium ion           1.0
 1.0   1    Ca     40.07980    Ca          0        calcium ion          2.0
 1.0   1    Ba     137.3300    Ba          0        barium ion           2.0
 1.0   1    Mg      24.3050    Mg          0        magnesium ion        2.0
 1.0   1    Sr      87.6200    Sr          0        strontium ion        2.0
 1.0   1    Pb     207.2000    Pb          0        lead ion             2.0
 1.0   1    Cl     35.45300    Cl          0        chloride ion        -1.0

#equivalence	cvff 

> Equivalence table for any variant of cvff 

!		         	  Equivalences
!                 -----------------------------------------
!Ver  Ref   Type  NonB     Bond    Angle    Torsion    OOP
!---- ---   ----  ----     ----    -----    -------    ----
 1.0   1    h     h        h        h        h         h   

#auto_equivalence	cvff_auto

!		         	  Equivalences
!                 -----------------------------------------                       
!Ver  Ref   Type  NonB Bond   Bond     Angle    Angle     Torsion   Torsion      OOP      OOP 
!                      Inct           End atom Apex atom End Atoms Center Atoms End Atom Center Atom
!---- ---   ----  ---- ------ ----  ---------- --------- --------- -----------  -------- ----------- 
 2.0  18    h     h     h     h_       h_       h_        h_        h_           h_       h_

#hbond_definition	cvff 

#morse_bond	cvff 

> E = D * (1 - exp(-ALPHA*(R - R0)))^2

!Ver  Ref     I     J          R0         D           ALPHA
!---- ---    ----  ----     -------    --------      -------
 2.3  23     no    o-        1.2178    140.2486      2.0000

#quadratic_bond	cvff 

> E = K2 * (R - R0)^2

!Ver  Ref     I     J          R0         K2    
!---- ---    ----  ----     -------    -------- 
 2.1  28      o*   h*        1.0000    553.9350
 2.1  28      oh   ho        1.0000    553.9350
 2.1  28     ohs   ho        1.0000    553.9350

#quadratic_angle	cvff 

> E = K2 * (Theta - Theta0)^2

!Ver  Ref     I     J     K       Theta0         K2        
!---- ---    ----  ----  ----    --------     -------
 1.0   1     h*    o*    h*      109.4700     45.7530

#torsion_1      cvff_auto

> E = Kphi * [ 1 + cos(n*Phi - Phi0) ]

!Ver  Ref     I     J      K      L           Kphi        n           Phi0
!---- ---    ----  ----   ----   ----      -------      ------     -------
 2.0  18     *     c_     n3n_   *          0.0500        3         0.
                                                                      
                                                                      

#out_of_plane	cvff_auto

> E = Kchi * [ 1 + cos(n*Chi - Chi0) ]

!Ver  Ref     I     J     K     L           Kchi        n           Chi0
!---- ---    ----  ----  ----  ----      -------      ------     -------
 2.0  18     *     c'_   *      *        10.0000        2       180.0000

#nonbond(12-6)	cvff 

@type A-B
@combination geometric

> E = Aij/r^12 - Bij/r^6
> where  Aij = sqrt( Ai * Aj )
>        Bij = sqrt( Bi * Bj )

!Ver  Ref     I           A             B 
!---- ---    ----    -----------   -----------
 1.0   1    st           12.3645       0.00954
 1.0   1    ao          196.1446       0.03230
 1.0   1    at           12.3645       0.00954
 1.0   1   mgo         1636.3265       0.07688
 1.0   1   mgh         1636.3265       0.07688
 1.0   1   cao         17814.73	       0.5987 
 1.0   1   cah         17624.076       0.595
 1.0   1   feo           702.54	       0.0504
 1.0   1   lio           112.01	       0.0201
 1.0   1    ob       629358.0000     625.50000
 1.0   1  obss      629358.0000     625.50000
 1.0   1  obts      629358.0000     625.50000
 1.0   1  obos      629358.0000     625.50000
 1.0   1   ohs      629358.0000     625.50000
 1.0   1    oh      629358.0000     625.50000
 1.0   1   oh-      629358.0000     625.50000
 1.0   1    o*      629358.0000     625.50000
 1.0   1    ho        0.00000001       0.00000
 1.0   1    h*        0.00000001       0.00000
 1.0   1    Na       14763.1719      87.65132
 1.0   1    K        754506.86       549.37
 1.0   1    Cs       3998193.96	    1264.63
 1.0   1    Ca      125966.6068     224.46969
 1.0   1    Ba      1799606.56	    582.25
 1.0   1    Mg      1369.00	     69.22
 1.0   1    Sr      1185860.37	    688.73
 1.0   1    Pb      861150.71	    638.08
 1.0   1    Cl      21081006.97     2905.31

#bond_increments        cvff

!Ver  Ref     I     J       DeltaIJ     DeltaJI
!---- ---    ----  ----     -------     -------
 2.3  23     no    o-        0.1684     -0.1684

**Comment by franzlang** [comment]
the original CLAYFF paper is J. Phys. Chem. B 2004, 108, 4, 1255–1266 (https://doi.org/10.1021/jp0363287)

**Comment by iermi** [comment]
Those are the typical forcefield files. The first part is describing atom types with connectivities between them and charges. Then when it goes to bonds, angles and dihedrals it specifies atom types which are forming those things and the constants (parameters for respective potentials). What you basically can learn from that is that not every oxygen is the same, not every hydrogen is the same etc. The first one is important to know when you create a starting structure and assign atom types. Other parts for bonds, angles and dihedrals are essential to know it they are exist for things which you have created. If a parameter is missing it has to be derived, because the simulation won't start without parameters. The similar idea about non-bonded parameters (for atoms which are not connected by a chemical bond).
